server/dals/expression-dal.js

```diff
@@ -96,16 +96,13 @@ const buildExpressionDataQuery = function (dataset, timepoints, genes) {
 
 
 const buildExpressionQuery = function (query) {
-    switch (query.type) {
-    case "DegradationRates":
-        return buildExpressionProductionDegradationRatesQuery("degradation_rate", query.genes);
-    case "ProductionRates":
-        return buildExpressionProductionDegradationRatesQuery("production_rate", query.genes);
-    case undefined:
-        // This is the original expression database call.
-        // Will restructure code so that it follows the standardized query type, query data structure that all
-        // other queries use
-        return buildExpressionDataQuery(query.dataset, query.timepoints, query.genes);
+    const expressionQueries = {
+        "DegradationRates": buildExpressionProductionDegradationRatesQuery("degradation_rate", query.genes),
+        "ProductionRates" : buildExpressionProductionDegradationRatesQuery("production_rate", query.genes),
+        "ExpressionData" : buildExpressionDataQuery(query.dataset, query.timepoints, query.genes)
```

Comment:
Suggest changing to functions in order to avoid evaluating all three even if you end up using just one

(this is a general pattern when you are doing data-driven implementation—watch out for when expressions are evaluated and generally try to avoid doing that until you know for sure that’s what you want)

Reply:
I fixed this